util/rate_limiter.cc

```diff
@@ -107,7 +107,10 @@ void GenericRateLimiter::Request(int64_t bytes, const Env::IOPriority pri) {
          (!queue_[Env::IO_LOW].empty() &&
             &r == queue_[Env::IO_LOW].front()))) {
       leader_ = &r;
-      timedout = r.cv.TimedWait(next_refill_us_);
+      int64_t nano_delta = next_refill_ns_ - env_->NowNanos();
+      nano_delta = nano_delta > 0 ? nano_delta : 0;
+      int64_t wait_util = env_->NowMicros() + nano_delta/1000;
+      timedout = r.cv.TimedWait(wait_util);
```

Comment:
Conversion back to microseconds defeats the purpose of the whole PR. Am I missing something?